scripts/linters/third_party_typings_linter.py

```diff
@@ -0,0 +1,151 @@
+    },
+    {
+        'name': 'Skulpt',
+        'manifest.json': 'skulpt-dist',
+        'type_defs_file_name': 'skulpt-defs-'
+    },
+    {
+        'name': 'Math Expressions',
+        'manifest.json': 'mathExpressions',
+        'type_defs_file_name': 'math-expressions-defs-'
+    },
+    {
+        'name': 'MIDI',
+        'manifest.json': 'midiJs',
+        'type_defs_file_name': 'midi-defs-'
+    },
+    {
+        'name': 'Wavesurfer',
+        'package.json': 'wavesurfer.js',
+        'type_defs_file_name': 'wavesurfer-defs-'
+    },
+    {
+        'name': 'Nerdamer',
+        'package.json': 'nerdamer',
+        'type_defs_file_name': 'nerdamer-defs-'
+    }
+]
+
+
+def check_third_party_libs_type_defs(verbose_mode_enabled):
+    """Checks the type definitions for third party libs
+    are up to date.
+    """
+    if verbose_mode_enabled:
+        python_utils.PRINT('Starting type defs check')
+        python_utils.PRINT('----------------------------------------')
+
+    with linter_utils.redirect_stdout(sys.stdout):
+        failed = False
+        summary_messages = []
+
+        with python_utils.open_file(_MANIFEST_JSON_FILE_PATH, 'r') as f:
+            manifest = json.load(f)['dependencies']['frontend']
+
+        with python_utils.open_file(_PACKAGE_JSON_FILE_PATH, 'r') as f:
+            package = json.load(f)['dependencies']
+
+        for third_party_lib in THIRD_PARTY_LIBS:
+            if 'manifest.json' in third_party_lib:
+                lib_version = (
+                    manifest[third_party_lib['manifest.json']]['version'])
+
+            if 'package.json' in third_party_lib:
+                lib_version = package[third_party_lib['package.json']]
+
+                if lib_version[0] == '^':
+                    lib_version = lib_version[1:]
+
+            files_in_typings_dir = os.listdir(
+                os.path.join(os.getcwd(), 'typings')
+            )
+
+            prefix_name = third_party_lib['type_defs_file_name']
+
+            files_with_prefix_name = []
+
+            for file_name in files_in_typings_dir:
+                if file_name.startswith(prefix_name):
+                    files_with_prefix_name.append(file_name)
+
+            if len(files_with_prefix_name) > 1:
+                python_utils.PRINT(
+                    'There are multiple type definitions for %s in the typings '
+                    'dir.' % third_party_lib['name']
+                )
+                python_utils.PRINT('')
+                failed = True
+            elif len(files_with_prefix_name) == 0:
+                python_utils.PRINT(
+                    'There are no type definitions for %s in the typings '
+                    'dir.' % third_party_lib['name']
+                )
+                python_utils.PRINT('')
+                failed = True
+            else:
+                type_defs_file_name = files_with_prefix_name[0]
+
+                type_defs_version = type_defs_file_name[len(prefix_name): -5]
```

Comment:
Can you explain the random `-5` here? (Also add a code comment)

Reply:
I found this -5 is for `.d.ts` if so can we use `len('.d.ts`) and add a comment on to explaining that all type-defs filename ends with this?

Reply:
done